### PR TITLE
Merge back Windows launcher code and add release workflow

### PR DESCRIPTION
**Pull request by neilcsmith-net**
This PR merges back the Windows native launcher code back into the main repo, and adds a GitHub workflow to build source and binary zips for release.

The first commit copies back files from https://github.com/apache/netbeans-native-launchers into their original locations.  Unfortunately, at present, history from the separate repo is lost.

The second commit fixes paths and adds a GitHub workflow based on the dlight and profiler workflows.  This builds a source and binary zip that we can release via dist.a.o in the same way as those other native binaries.

The workflow uses the same technique of building a source zip keeping the native libraries in their existing relative path locations.  Separate isolated steps build the binary files and binary zip from that source zip (ensuring completeness of the sources).

Artefacts still need testing.  A second PR will need generating to use the staged artefacts for test build and vote (see process used for other binaries).

As well as pending changes, we'll need to consider versioning in `versions.h` and possibly whether that can be configured via environment variables that can be set via GH workflow??

### CODE REVIEW

**Review by mbien — APPROVED** [review]
workflow looks good.

Not importing the history is probably fine in this case.

**Comment by mbien** [comment]
wondering if a java program compiled to native image could launch NB, analog to how the mvnd CLI client works which communicates with the daemon.

**Comment by neilcsmith-net** [comment]
@mbien have had that thought, too.  Some easier replacement for the Windows launchers would be good to consider in a future update.  In some ways, having the bulk of the logic in a script form could also be good to explore.  The macOS Swift launcher is just a tiny stub that executes the shell script.  Anyway, let's follow up on that on mailing list or Slack.  Important thing initially is to find an easier way to include the existing functionality.

File: .github/workflows/native-binary-build-launcher.yml
Comment (neilcsmith-net) [reply .github/workflows/native-binary-build-launcher.yml:38]: Not sure whether to uncomment or remove.  It makes developing a pain.  Needed to comment out to get the workflow to trigger on my own repo while working on it.  Not sure if we can limit the push branch filter to only upstream and not forks?  Otherwise, think we can lose the filters on all the native workflows.

Comment (mbien) [reply .github/workflows/native-binary-build-launcher.yml:38]: would workflow dispatch work for your usecase as trigger? I think you can even dispatch from gh CLI these days.

We have committers using branches on the main repo still which is the original reason why I added those. (see top of https://github.com/apache/netbeans/actions for a recent example)

Comment (neilcsmith-net) [reply .github/workflows/native-binary-build-launcher.yml:38]: Workflow dispatch requires the workflow to be in the default branch.  I realise the reason for having the branch filter on the main workflow.  I think having branch _or_ path filter on push should be enough.

Comment (neilcsmith-net) [reply .github/workflows/native-binary-build-launcher.yml]: I've removed the commented code for now.  We can add back if paths prove to not be enough in practice.

**Comment by neilcsmith-net** [comment]
Just updated the second (IDE) version file to match the one in bootstrap.  Need to look at how we inject these from the action ideally.

Currently set to `101.1.0.0` because that is higher than existing and no way to be confused with IDE / platform versions (for the foreseeable future!)

File: .github/workflows/native-binary-build-launcher.yml
Comment (mbien) [reply .github/workflows/native-binary-build-launcher.yml]: ok, just be aware that this would be the only workflow without branch filter now.

File: platform/o.n.bootstrap/launcher/windows/version.h
Comment (mbien) [inline platform/o.n.bootstrap/launcher/windows/version.h:23]: there might be more versions to change: https://issues.apache.org/jira/browse/NETBEANS-1590

Comment (neilcsmith-net) [reply platform/o.n.bootstrap/launcher/windows/version.h:23]: Yes, thanks, I noticed those earlier.  I've limited to the ones that @ebarboni updated in the other repo.  I assume that issue still applies.  We really need to find a way to centralize and configure these!

Comment (mbien) [reply platform/o.n.bootstrap/launcher/windows/version.h:23]: maybe some of those don't need to be versions at all. Not sure what happens if the exe has just `Apache NetBeans` in the manifest.

Comment (ebarboni) [reply platform/o.n.bootstrap/launcher/windows/version.h:23]: I'm not windows dev expert, I've no idea on how the manifest should be and what its quality should be to be win friendly. 

We may need to add Apache in front of NetBeans.

**Review by ebarboni — APPROVED** [review]
LGTM

File: platform/o.n.bootstrap/launcher/windows/version.h
Comment (mbien) [reply platform/o.n.bootstrap/launcher/windows/version.h:23]: looks like its required:
https://learn.microsoft.com/en-us/windows/win32/menurc/stringfileinfo-block

but its a string. So it could be "LauncherV1.1" or something arbitrary to not make the NB version connection.

Comment (neilcsmith-net) [reply platform/o.n.bootstrap/launcher/windows/version.h:23]: I've updated the version in the manifest files and added Apache in front of NetBeans in the component and name fields.

There are product and file versions in the stringfileinfo-block which appear to be required.  Usage of `BUILD_ID` also needs checking.  The version in the manifest files also appears to be required.  https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#assemblyIdentity  And that has a stricter format.  We'd also need to work out how to inject a version in there.

Hopefully this update is enough for now - we'll need to look at how we get pending changes in and release in time for NB21.

**Comment by neilcsmith-net** [comment]
I'll look to merge this tomorrow unless anyone has further comments or wants to request changes.

cc/ @jlahoda if we can then bring the changes suggested in the separate repo on top of this, we should be able to proceed with a launcher release with those changes prior to freeze.

**Comment by mbien** [comment]
I think this would close https://issues.apache.org/jira/browse/NETBEANS-1590 too once merged. I am mostly posting this because i would otherwise forget about it.

**Comment by neilcsmith-net** [comment]
@mbien in the short term, yes.  We also need to not forget about finding a better way to manage versioning so that we don't let that issue recur.  And also, how we progress from `101.1.0.0` - in my mind, incidentally, I used `101` as a prefix ahead of the IDE and anything used before, with the thought we'd then do `101.2...`, `101.3...` etc.

**Comment by mbien** [comment]
should https://github.com/apache/netbeans-native-launchers be archived?

**Pull request merged**
